src/lib/defs.h

```diff
@@ -136,6 +136,26 @@
         }                                                                             \
     } while (/*CONSTCOND*/ 0)
 
+#define EXPAND_ARRAY_EX(str, arr, num)                                                \
+    do {                                                                              \
+        if (str->arr##c == str->arr##vsize) {                                         \
+            void *   __newarr;                                                        \
+            char *   __newarrc;                                                       \
+            unsigned __newsize;                                                       \
+            __newsize = str->arr##vsize + num;                                        \
+            if ((__newarrc = __newarr =                                               \
+                   realloc(str->arr##s, __newsize * sizeof(*str->arr##s))) == NULL) { \
+                (void) fprintf(stderr, "EXPAND_ARRAY - bad realloc\n");               \
+            } else {                                                                  \
+                (void) memset(&__newarrc[str->arr##vsize * sizeof(*str->arr##s)],     \
+                              0x0,                                                    \
+                              (__newsize - str->arr##vsize) * sizeof(*str->arr##s));  \
+                str->arr##s = __newarr;                                               \
+                str->arr##vsize = __newsize;                                          \
+            }                                                                         \
+        }                                                                             \
+    } while (/*CONSTCOND*/ 0)
+
```

Comment:
Maybe do not duplicate code and rewrite `EXPAND_ARRAY` to use a new macros?

Reply:
I did this PR as non-breaking so added this macro instead of rewriting. Actually, most likely, we'll get rid of this define at some point in future at all.